### PR TITLE
LIS-PREINSTALL-DISK-SIZE-VERIFICATION: Implemented Dynamic size check

### LINKED ISSUE #333: Test case LIS-PREINSTALL-DISK-SIZE-VERIFICATION issue
Still available 70 M before install LIS for negative case
LIS can be installed if have more than 468K space
installing package kmod-microsoft-hyper-v-4.3.3-20190608.x86_64 needs 468KB on the / filesystem

```
Filesystem     1M-blocks  Used Available Use% Mounted on
/dev/sda1          30109 28504        70 100% /
tmpfs               1720     0      1720   0% /dev/shm
/dev/sdb1           6927    16      6553   1% /mnt/resource
```

### PR DESCRIPTION
**Pull request by DivyaJyothiKasam**
Fixes #333 

Implemented Dynamic size check before LIS installation instead of  static method.

### CODE REVIEW

**Comment by DivyaJyothiKasam** [comment]
Same partition:

Execution log:
[LISAv2 Test Results Summary]
Test Run On           : 06/21/2019 11:08:39
ARM Image Under Test  : OpenLogic : CentOS : 6.9 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:9
   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 LIS_DEPLOY           LIS-PREINSTALL-DISK-SIZE-VERIFICATION                                             PASS                    4 
	

Script logs:
DISTRO PACKAGE NAME: RPMS DISTRO VERSION: 69
boot_partition /dev/sda1 - root_partition /dev/sda1
ramdisk_size_factor 1 root_part_avail_space 27512340480 boot_part_avail_space 27512340480
lib_module_required_space 10318726 ramdisk_required_space 25410664
Fri Jun 21 11:14:54 2019 : boot and root share the same partition. We will create only one file
single_partition_file_size 27308838930
Fri Jun 21 11:14:54 2019 : /lib/modules/file.out created with size 27308838930
singe_partition_created_file_size 27308838930
Fri Jun 21 11:14:55 2019 : singe_partition_created_file_size : 27308838930
  File: "/"
    ID: cc091f246ec91458 Namelen: 255     Type: ext2/ext3
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 7707903    Free: 442884     Available: 49681
Inodes: Total: 1966080    Free: 1908778
Fri Jun 21 11:14:55 2019 : Filesystem     1M-blocks  Used Available Use% Mounted on
/dev/sda1          30109 28379       195 100% /
tmpfs               1720     0      1720   0% /dev/shm
/dev/sdb1           6927    16      6553   1% /mnt/resource
Fri Jun 21 11:14:55 2019 : created
Removing Hyper-V daemons
Removing Hyper-V Tools
Invoking release specific install file in directory RPMS69
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                ##################################################
kmod-microsoft-hyper-v      ##################################################
Preparing...                ##################################################
microsoft-hyper-v-debuginfo ##################################################
microsoft-hyper-v           ##################################################
Saving old initramfs
Installing new initramfs
Adding KVP Daemon to Chkconfig....
Starting KVP Daemon....
Adding VSS Daemon to Chkconfig....
Starting VSS Daemon....
Adding FCOPY Daemon to Chkconfig....
Starting FCOPY Daemon....
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.
Fri Jun 21 11:16:23 2019 : Cleanup completed.
Fri Jun 21 11:16:24 2019 : Found 3 hyper-v packages installed
Fri Jun 21 11:16:24 2019 : Now removing LIS...
Uninstalling the Linux Integration Services for Microsoft Hyper-V...
Removing Package..
Stopping KVP Daemon....
Deleting KVP Daemon from Chkconfig....
Stopping VSS Daemon....
Deleting VSS Daemon from Chkconfig....
Stopping FCOPY Daemon....
Deleting FCOPY Daemon from Chkconfig....
Create and install initramfs without Hyper-V drivers
Linux Integration Services for Hyper-V has been removed.
warning: /etc/depmod.d/hyperv.conf saved as /etc/depmod.d/hyperv.conf.rpmsave
Uninstalled Linux Integration Services for Microsoft Hyper-V. Please reboot your system.
Fri Jun 21 11:17:48 2019 : Found 0 hyper-v packages installed
Fri Jun 21 11:17:48 2019 : Test 'positive' : PASS
Fri Jun 21 11:17:48 2019 : Cleanup completed.
DISTRO PACKAGE NAME: RPMS DISTRO VERSION: 69
boot_partition /dev/sda1 - root_partition /dev/sda1
ramdisk_size_factor 1 root_part_avail_space 27462971392 boot_part_avail_space 27462971392
lib_module_required_space 10318726 ramdisk_required_space 25412145
negative: root_part_required_space : 101751515 boot_part_required_space: 13230360
Fri Jun 21 11:17:48 2019 : boot and root share the same partition. We will create only one file
single_partition_file_size 27361219877
Fri Jun 21 11:17:48 2019 : /lib/modules/file.out created with size 27361219877
singe_partition_created_file_size 27361219877
Fri Jun 21 11:17:48 2019 : singe_partition_created_file_size : 27361219877
  File: "/"
    ID: cc091f246ec91458 Namelen: 255     Type: ext2/ext3
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 7707903    Free: 418043     Available: 24840
Inodes: Total: 1966080    Free: 1908775
Fri Jun 21 11:17:48 2019 : Filesystem     1M-blocks  Used Available Use% Mounted on
/dev/sda1          30109 28477        98 100% /
tmpfs               1720     0      1720   0% /dev/shm
/dev/sdb1           6927    16      6553   1% /mnt/resource
Fri Jun 21 11:17:48 2019 : created
LIS installation aborted due to insufficient space
         /lib/modules            available: 101744640 required: 193017271 bytes
Fri Jun 21 11:17:48 2019 : Cleanup completed.
Fri Jun 21 11:17:49 2019 : Found 0 hyper-v packages installed
Fri Jun 21 11:17:49 2019 : Test 'negative' : PASS
Fri Jun 21 11:17:49 2019 : Cleanup completed.
Fri Jun 21 11:17:49 2019 : Cleanup completed.

**Comment by DivyaJyothiKasam** [comment]
Different partition:

Execution log:
[LISAv2 Test Results Summary]
Test Run On           : 06/21/2019 11:18:59
ARM Image Under Test  : RedHat : RHEL : 7.5 : Latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:7
   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 LIS_DEPLOY           LIS-PREINSTALL-DISK-SIZE-VERIFICATION                                             PASS                 4.95 


Script logs:
DISTRO PACKAGE NAME: RPMS DISTRO VERSION: 75
boot_partition /dev/sda1 - root_partition /dev/sda2
ramdisk_size_factor 2 root_part_avail_space 32004767744 boot_part_avail_space 416038912
lib_module_required_space 14832683 ramdisk_required_space 19371127
Fri Jun 21 11:22:28 2019 : boot and modules do not share the same partition.
file size - root partition: 31783420647 boot partition: 395619209
Fri Jun 21 11:22:28 2019 : /lib/modules/file.out created with size 31783420647
Fri Jun 21 11:22:28 2019 : /boot/file.out created with size 395619209
  File: "/"
    ID: 80200000000 Namelen: 255     Type: xfs
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 8256319    Free: 54038      Available: 54038
Inodes: Total: 480176     Free: 432397
Fri Jun 21 11:22:28 2019 : Filesystem     1M-blocks  Used Available Use% Mounted on
/dev/sda2          32252 32041       212 100% /
devtmpfs            1705     0      1705   0% /dev
tmpfs               1716     0      1716   0% /dev/shm
tmpfs               1716     9      1707   1% /run
tmpfs               1716     0      1716   0% /sys/fs/cgroup
/dev/sda1            497   478        20  97% /boot
/dev/sdb1           6927  2080      4473  32% /mnt/resource
tmpfs                344     0       344   0% /run/user/1000
Fri Jun 21 11:22:28 2019 : created
Invoking release specific install file in directory RPMS75
Removing Hyper-V daemons
Removing Hyper-V Tools
Installing the Linux Integration Services for Microsoft Hyper-V...
Preparing...                          ########################################
Updating / installing...
kmod-microsoft-hyper-v-4.3.3.rc1-20190########################################
microsoft-hyper-v-4.3.3.rc1-20190621  ########################################
Saving old initramfs
Installing new initramfs
Starting KVP Daemon....
Starting VSS Daemon....
Starting FCOPY Daemon....
microsoft-hyper-v-debuginfo-4.3.3.rc1-########################################
 Linux Integration Services for Hyper-V has been installed. Please reboot your system.
Fri Jun 21 11:24:50 2019 : Cleanup completed.
Fri Jun 21 11:24:51 2019 : Found 3 hyper-v packages installed
Fri Jun 21 11:24:51 2019 : Now removing LIS...
Uninstalling the Linux Integration Services for Microsoft Hyper-V...
Removing Package..
Stopping KVP Daemon....
Stopping FCOPY Daemon....
Stopping VSS Daemon....
Create and install initramfs without Hyper-V drivers
Linux Integration Services for Hyper-V has been removed.
warning: /etc/depmod.d/hyperv.conf saved as /etc/depmod.d/hyperv.conf.rpmsave
Uninstalled Linux Integration Services for Microsoft Hyper-V. Please reboot your system.
Fri Jun 21 11:26:24 2019 : Found 0 hyper-v packages installed
Fri Jun 21 11:26:24 2019 : Test 'positive' : PASS
Fri Jun 21 11:26:24 2019 : Cleanup completed.
DISTRO PACKAGE NAME: RPMS DISTRO VERSION: 75
boot_partition /dev/sda1 - root_partition /dev/sda2
ramdisk_size_factor 2 root_part_avail_space 31990820864 boot_part_avail_space 398286848
lib_module_required_space 14832683 ramdisk_required_space 18482698
negative: root_part_required_space : 109785119 boot_part_required_space: 9765637
Fri Jun 21 11:26:25 2019 : boot and modules do not share the same partition.
file size - root partition: 31881035745 boot partition: 388521211
Fri Jun 21 11:26:25 2019 : /lib/modules/file.out created with size 31881035745
Fri Jun 21 11:26:25 2019 : /boot/file.out created with size 388521211
  File: "/"
    ID: 80200000000 Namelen: 255     Type: xfs
Block size: 4096       Fundamental block size: 4096
Blocks: Total: 8256319    Free: 26802      Available: 26802
Inodes: Total: 262480     Free: 214691
Fri Jun 21 11:26:25 2019 : Filesystem     1M-blocks  Used Available Use% Mounted on
/dev/sda2          32252 32147       105 100% /
devtmpfs            1705     0      1705   0% /dev
tmpfs               1716     0      1716   0% /dev/shm
tmpfs               1716     9      1707   1% /run
tmpfs               1716     0      1716   0% /sys/fs/cgroup
/dev/sda1            497   488        10  99% /boot
/dev/sdb1           6927  2080      4473  32% /mnt/resource
tmpfs                344     0       344   0% /run/user/1000
Fri Jun 21 11:26:25 2019 : created
LIS installation aborted due to insufficient space
         /lib/modules            available: 109780992 required: 209084479 bytes
Fri Jun 21 11:26:25 2019 : Cleanup completed.
Fri Jun 21 11:26:26 2019 : Found 0 hyper-v packages installed
Fri Jun 21 11:26:26 2019 : Test 'negative' : PASS
Fri Jun 21 11:26:26 2019 : Cleanup completed.
Fri Jun 21 11:26:26 2019 : Cleanup completed.

File: Testscripts/Linux/Test_LIS_disk_check.sh [hunk 1]
```diff
@@ -98,71 +98,81 @@ Install_Uninstall_LIS() {
...
-		CreateFile "${boot_directory_target_size}M" /boot/file.out
-		boot_created_file_size=$(ls -l --block-size=M /boot/file.out | awk '{print $5}'  | tail -1)
-		if [[ "$module_created_file_size" == "${modules_directory_target_size}M"
-			&& "$boot_created_file_size" == "${boot_directory_target_size}M" ]]; then
+		LogMsg "boot and modules do not share the same partition."
+		single_partition_file_size=$(expr $root_part_avail_space - $root_part_required_space)
+		single_partition_file_size_boot=$(expr $boot_part_avail_space - $boot_part_required_space)
+		echo "file size - root partition: $single_partition_file_size boot partition: $single_partition_file_size_boot"
+		CreateFile "${single_partition_file_size}" /lib/modules/file.out
+		singe_partition_created_file_size=$(ls -l /lib/modules/file.out | awk '{print $5}'  | tail -1)
```
Comment (chvalean) [inline Testscripts/Linux/Test_LIS_disk_check.sh:164]: typo in single

File: Testscripts/Linux/Test_LIS_disk_check.sh [hunk 2]
```diff
@@ -98,71 +98,81 @@ Install_Uninstall_LIS() {
...
+		boot_created_file_size=$(ls -l /boot/file.out | awk '{print $5}'  | tail -1)
+		if [[ "$single_partition_file_size" == "${singe_partition_created_file_size}"
+			&& "$single_partition_file_size_boot" == "${boot_created_file_size}" ]]; then
 			file_status="created"
 		else
 			file_status="failed"
 		fi
 	fi
+
+	stat -f  /
```
Comment (chvalean) [inline Testscripts/Linux/Test_LIS_disk_check.sh:175]: what is the role of this command?
output is not saved anywhere, and it doesn't look to do any relevant operation?

**Review by chvalean — CHANGES_REQUESTED** [review]

**Comment by chvalean** [comment]
> Implemented Dynamic size check before LIS installation instead of static method

@DivyaJyothiKasam can you please explain the changes here a little bit?
What was wrong in original implementation, and how these changes are making the test more stable/reliable?

**Comment by LiliDeng** [comment]
@chvalean  see #333

**Comment by chvalean** [comment]
> @chvalean see #333

PR comment updated to link to that issue, thank you!

**Comment by LiliDeng** [comment]
@DivyaJyothiKasam Please run tests based on latest code, we hit failures against  OpenLogic : CentOS : 6.9 : Latest with your code.

**Review by chvalean — COMMENTED** [review]
removing myself from review.

This test-case changes must be reviewed by @iamshital and @johnsongeorge-w

**Comment by johnsongeorge-w** [comment]
Closing this PR as Sireesha will start working on it. Same modifications will be cherry-picked and review comments will be addressed. After testing Sireesha will raise a new PR.

**Pull request closed without merging**